src/overlays/actors/ovl_Dm_Char02/z_dm_char02.c

```diff
@@ -28,30 +28,151 @@ const ActorInit Dm_Char02_InitVars = {
+    switch (globalCtx->csCtx.frames) {
+        case 0x5F:
+            Actor_PlaySfxAtPos(actor, 0x292AU);
+            return;
+        case 0x65:
+        case 0x69:
+        case 0x70:
+            Actor_PlaySfxAtPos(actor, 0x292BU);
+            return;
+    }
+}
+
+void func_80AAAF2C(DmChar02* this, GlobalContext* globalCtx) {
+    if ((globalCtx->csCtx.state != 0) && (globalCtx->sceneNum == SCENE_OKUJOU) &&
+        (globalCtx->csCtx.currentCsIndex == 1)) {
+        func_80AAAECC(&this->actor, globalCtx);
+    }
+}
+
+void DmChar02_Init(Actor* thisx, GlobalContext* globalCtx) {
+    DmChar02* this = THIS;
+
+    // items[0] is the ocarina
+    if (gSaveContext.save.inventory.items[0] == 0xFF) {
+        this->unk2E0 = 0;
+        this->actor.targetArrowOffset = 3000.0f;
+        ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 24.0f);
+        SkelAnime_InitFlex(globalCtx, &this->skelAnime, &gClockTowerOcarinaOfTimeSkel, NULL, NULL, NULL, 0);
+        DmChar02_ChangeAnimationByInfo(&this->skelAnime, sAnimations, 0);
+        Actor_SetScale(&this->actor, 0.01f);
+        this->actionFunc = func_80AAB04C;
+        return;
+    }
+    Actor_MarkForDeath(&this->actor);
+}
+
+void DmChar02_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
+
+void func_80AAB04C(DmChar02* this, GlobalContext* globalCtx) {
+    u8 sp2F;
+    s32 actionIndex;
+
+    sp2F = 1;
+    if (Cutscene_CheckActorAction(globalCtx, 0x83U) != 0) {
+        actionIndex = Cutscene_GetActorActionIndex(globalCtx, 0x83U);
+        if (globalCtx->csCtx.frames == globalCtx->csCtx.actorActions[actionIndex]->startFrame) {
+            switch (globalCtx->csCtx.actorActions[actionIndex]->action) {
+                default:
+                    this->unk2E0 = 0;
+                    sp2F = 0;
+                    break;
+                case 1:
+                    this->unk2E0 = 0;
+                    break;
+                case 2:
+                    this->unk2E0 = 1;
+                    break;
+                case 4:
+                    this->unk2E0 = 3;
+                    break;
+            }
+            if (sp2F != 0) {
+                DmChar02_ChangeAnimationByInfo(&this->skelAnime, &sAnimations[this->unk2E0], 0U);
+            }
+        }
+        Cutscene_ActorTranslateAndYaw(&this->actor, globalCtx, actionIndex);
+    }
+    if (Animation_OnFrame(&this->skelAnime, this->skelAnime.endFrame) != 0) {
+        if (this->unk2E0 == 1) {
+            this->unk2E0++;
+            DmChar02_ChangeAnimationByInfo(&this->skelAnime, &sAnimations[this->unk2E0], 0U);
+        }
+    }
+}
+
+void DmChar02_Update(Actor* thisx, GlobalContext* globalCtx) {
+    DmChar02* this = THIS;
+    SkelAnime_Update(&this->skelAnime);
+    this->unk2F0 = this->unk2F0;
+    this->actionFunc(this, globalCtx);
+    if (Actor_HasParent(&this->actor, globalCtx) == 0) {
+        Actor_PickUp(&this->actor, globalCtx, 0x4C, 30.0f, 80.0f);
```

Comment:
This should use the GI enum, in this case you have to add an entry for the ocarina.